### ISSUE TITLE
Kinect + Wiimote instead of Leap Motion

### ISSUE THREAD

**Issue by BritishLad** [ajayyy/Wiimote-FreePIE-VR-Controls#7]
sorry if i am late to this.

but im getting the errors i have attached below

![errors](https://user-images.githubusercontent.com/35122669/34592546-cab779f8-f1bc-11e7-915f-198b6ae31460.jpg)

sorry about the second screen,

would love some help 

thank you

**Comment by ajayyy** [comment]
Have you tried another file in the repo? I know some of them don't work for some people.

**Comment by BritishLad** [comment]
do you know where i would find one?

i was searching around and couldn't find anything

**Comment by ajayyy** [comment]
Inside my repository, there are other files you can try. 

wiimote_leap_vive.py
wiimote_leap_vive_2.py
wiimote_leap_vive_nogyro.py

https://github.com/ajayyy/Wiimote-FreePIE-VR-Controls

Are you using leap motion?

**Comment by BritishLad** [comment]
no I'm not using leap motion

sorry i should have said.

**Comment by ajayyy** [comment]
ok, because the wiimote_vive.py also uses leap motion, that's probably why you are getting the error.

I will help in a few hours, gonna be busy sorry.

**Comment by BritishLad** [comment]
Its all good mate. 

ill try the other files and remove the leap parts.

thanks for the help

**Comment by Vininess** [comment]
@BritishLad I get the same error on first time, just grab the latest version of freepie (1.10.692.0)

Some versions have some issues with DolphiiMote.dll, you can substitute this dll file too

**Comment by BritishLad** [comment]
@Vininess thx mate i substituted in a different DolphiiMote.dll and it worked

however Im not getting any feed back on the wiimotes

**Comment by ajayyy** [comment]
@BritishLad I would like to know what you are planning on doing, what are you going to use this with, how do you plan to do the positions of the hands, is it going to be the same as @Vininess ?

**Comment by BritishLad** [comment]
@ajayyy I will be doing the same as @Vininess yes, by using kinect, however if that doesnt work correctly as i hope i will use ps eye cameras and using different colour leds to track the wii motes

**Comment by ajayyy** [comment]
@BritishLad ok, so you guys just want button presses, I'll whip up a quick script for that for you to test and see if it works

**Comment by ajayyy** [comment]
@BritishLad and @Vininess Ok try wiimote_vive_only_buttons.py

**Comment by BritishLad** [comment]
@ajayyy yeah all works. thx for that.

however im not to sure if the kinect will work, as i cant find a tracker yet

**Comment by ajayyy** [comment]
@BritishLad I know you can use https://github.com/sheepsskullcity/opentrackkinect for positional tracking, but not controller positions.
 

And @Vininess was saying Driver4VR can do it

**Comment by BritishLad** [comment]
@ajayyy yes it can however it is A) a time limited program B) the tracking is sketchy

**Comment by ajayyy** [comment]
@BritishLad oh, ok. You can try opentrackkinect for positional tracking for now

**Comment by ajayyy** [comment]
@BritishLad have you tried https://github.com/jakers1403/Kinect-Plugin-Freepie
  
Edit: looks like I never got that working, but you could try it

**Comment by BritishLad** [comment]
I followed the steps to install it, however i think I'm having the same problem you did. as free pie wont open

**Comment by ajayyy** [comment]
aw that sucks

**Comment by BritishLad** [comment]
@ajayyy do u think there is a way to trick psmove into thinking the kinect is one of its webcams?

**Comment by ajayyy** [comment]
I remember them saying that they were going to add normal web cam support in the future, but they probably have not yet.

**Comment by BritishLad** [comment]
@ajayyy hey i got it working and however how would i be able to translate this into steamvr for tracking?

would i substitute it for the hydra positions?

**Comment by ajayyy** [comment]
what did you get working for positions?

**Comment by BritishLad** [comment]
@ajayyy i got tracking for the kinect to work, as well i used the acceleration of the wiimotes for the roll and pitch as i don't have motion plus. However how would i make it so the controllers stay in-front of my headset as the are stuck to one orientation not following the motion of the headset.

ill be busy for the next few hours 12 - 15 so ill come back on a few

**Comment by BritishLad** [comment]
@ajayyy any idea on how I would get it to stay with the headset?

**Comment by ajayyy** [comment]
That's what https://github.com/ajayyy/Wiimote-FreePIE-VR-Controls/blob/master/wiimote_leap_vive_2.py#L186 (line 186 and farther) does, you'd have to reverse engineer that and figure out how to that.

This assumes the rotation data can be pulled from freetrack. Are you using vridge, I'm not 100% sure if vridge puts the data to freetrack.

**Comment by BritishLad** [comment]
i have now made the hands stick to where my headset is positioned by using the kinect to track my head then taking it away from the hand positions, so my hands do move however i cant move the yaw of my hands. 

as my wii motes dont have motions plus :/

any other way u know?

**Comment by ajayyy** [comment]
No, I think you would somehow need to track your hands with the kinect

**Comment by BritishLad** [comment]
am i better off getting the motion plus modules?

**Comment by Vininess** [comment]
@BritishLad with the wiimotion plus you got the yaw, because the motion plus adds more 3 giros to get a better respons... but you need to recalibrate every tome manually, because they don't have an magnetometer like in the psmove, the wii remote use the sensor bar to recalibrate

I try to figure out an idea to auto recalibrate the controller

**Comment by Vininess** [comment]
https://www.youtube.com/watch?v=jPP8XQRqlLA
This video can be a solution, this guy create a code for less drifting

However i have two ideas
1: Make a code like in the video
2: buy a cheep magnetometer on aliexpress and soldering on wiiremote

**Comment by ajayyy** [comment]
That's a cool idea

**Comment by Vininess** [comment]
@ajayyy Do you mean the code or the soldering?

**Comment by ajayyy** [comment]
The soldering, I'm not sure how it would work but it would be cool. Maybe doing something like the original project this is forked from did https://github.com/fmaurer/FreePIE-VR-Controls

They made a custom controller out of an arduino.

**Comment by Vininess** [comment]
I have the tools and i know how to soldering, but i try to figure out where i have to solder the magnetometer to get the effect

**Comment by BritishLad** [comment]
@Vininess I have just ordered the motion plus adapters off of Amazon and i have completed/coded the kinect for my hand positions. When i have completed the code ill send @ajayyy and @Vininess a copy, if you wanna try it out.

**Comment by Vininess** [comment]
@BritishLad thanks for the help, i know a little how to code, but i know to soldering and math and physics stuffs, if you need some help in that parts cont on me

I bought the kinect today, but i discovered that i need an extension/power supply to plug on PC... so i back to the point 0

@BritishLad you coded the kinect using the freepie or another stuff? i heard/read, that the Driver4VR is quite good, but 20 euros converting to BRL, i will pay 80 BRL is quiet expensive... on kinect i pay 50 BRL, so the software is 2x the price, i want to try some alternatives

Thanks, when you send i check and feedback to you

**Comment by BritishLad** [comment]
first u will need this for ur FreePie https://github.com/sheepsskullcity/opentrackkinect this allows the plugin for FreePie to use the input/tracking. 

The Code is a mix of @ajayyy wiimote controller buttons only and the kinect code.

```
def vive_controller1():

	finetuning = 23
	
	hydra[1].x = kinectPlugin.handLeftX*finetuning - kinectPlugin.headX
	hydra[1].y = kinectPlugin.handLeftY*finetuning - kinectPlugin.headY
	hydra[1].z = kinectPlugin.handLeftZ*30 - kinectPlugin.headZ
	
	hydra[1].pitch = -20
	
	Lone = wiimote[1].buttons.button_down(WiimoteButtons.DPadUp) or wiimote[1].buttons.button_down(WiimoteButtons.DPadDown) or wiimote[1].buttons.button_down(WiimoteButtons.DPadLeft) or wiimote[1].buttons.button_down(WiimoteButtons.DPadRight)
	Lfour = wiimote[1].buttons.button_down(WiimoteButtons.Plus)
	Lstart = wiimote[1].buttons.button_down(WiimoteButtons.Home)
	Lbumper = wiimote[1].buttons.button_down(WiimoteButtons.A)
	
	hydra[1].one = Lone
	hydra[1].four = Lfour
	hydra[1].start = Lstart
	hydra[1].bumper = Lbumper
	hydra[1].joybutton = Lone
	if wiimote[1].buttons.button_down(WiimoteButtons.DPadRight):
		hydra[1].joyx = 1
	elif wiimote[1].buttons.button_down(WiimoteButtons.DPadLeft):
		hydra[1].joyx = 0
	if wiimote[1].buttons.button_down(WiimoteButtons.DPadUp):
		hydra[1].joyy = 1
	elif wiimote[1].buttons.button_down(WiimoteButtons.DPadDown):
		hydra[1].joyy = 0
	
	hydra[0].x = kinectPlugin.handRightX*finetuning - kinectPlugin.headX
	hydra[0].y = kinectPlugin.handRightY*finetuning - kinectPlugin.headY
	hydra[0].z = kinectPlugin.handRightZ*30 - kinectPlugin.headZ
	
	hydra[0].pitch = -20
		
	Rone = wiimote[0].buttons.button_down(WiimoteButtons.DPadUp) or wiimote[0].buttons.button_down(WiimoteButtons.DPadDown) or wiimote[0].buttons.button_down(WiimoteButtons.DPadLeft) or wiimote[0].buttons.button_down(WiimoteButtons.DPadRight)
	Rfour = wiimote[0].buttons.button_down(WiimoteButtons.Plus)
	Rstart = wiimote[0].buttons.button_down(WiimoteButtons.Home)
	Rbumper = wiimote[0].buttons.button_down(WiimoteButtons.A)
	
	hydra[0].one = Rone
	hydra[0].four = Rfour
	hydra[0].start = Rstart
	hydra[0].bumper = Rbumper
	hydra[0].joybutton = Rone
	if wiimote[0].buttons.button_down(WiimoteButtons.DPadRight):
		hydra[0].joyx = 1
	elif wiimote[0].buttons.button_down(WiimoteButtons.DPadLeft):
		hydra[0].joyx = 0
	if wiimote[0].buttons.button_down(WiimoteButtons.DPadUp):
		hydra[0].joyy = 1
	elif wiimote[0].buttons.button_down(WiimoteButtons.DPadDown):
		hydra[0].joyy = 0
	
	joyLi = 0
	joyRi = 1
		
	diagnostics.watch(wiimote[joyLi].buttons.button_down(WiimoteButtons.B))
	diagnostics.watch(wiimote[joyRi].buttons.button_down(WiimoteButtons.B))
	
	diagnostics.watch(wiimote[joyLi].buttons.button_down(WiimoteButtons.A))
	diagnostics.watch(wiimote[joyRi].buttons.button_down(WiimoteButtons.A))
	diagnostics.watch(wiimote[joyRi].buttons.button_down(WiimoteButtons.DPadUp))
	diagnostics.watch(wiimote[joyLi].buttons.button_down(WiimoteButtons.DPadDown))
	
	diagnostics.watch(kinectPlugin.handLeftX)
	diagnostics.watch(kinectPlugin.handLeftY)
	diagnostics.watch(kinectPlugin.handLeftZ)
	
	diagnostics.watch(kinectPlugin.handRightX)
	diagnostics.watch(kinectPlugin.handRightY)
	diagnostics.watch(kinectPlugin.handRightZ)

def hydra_init():
	hydra[0].pitch = 0
	hydra[0].yaw = 0
	hydra[0].roll = 0
	hydra[0].x = 0
	hydra[0].y = 0
	hydra[0].z = 0

	hydra[1].pitch = 0
	hydra[1].yaw = 0
	hydra[1].roll = 0
	hydra[1].x = 0
	hydra[1].y = 0
	hydra[1].z = 0
	
def vive_init():
	global posx
	global posy
	global posz
	posx = 0
	posy = 0
	posz = 0
	global orrpitch
	global orryaw
	global orrroll
	orrpitch = 0
	orryaw = 0
	orrroll = 0
	
global s

if starting:
	vive_init()
	s = 1
	hydra_init()

if s == 1:
	vive_controller1()
```

**Comment by Vininess** [comment]
@BritishLad i will need the ping pong glowing to kinect tack?

**Comment by BritishLad** [comment]
No need for that, works without it

**Comment by Vininess** [comment]
@BritishLad i see, thanks, tomorrow i will try and feedback

**Comment by ajayyy** [comment]
@BritishLad Wow that's cool nice job. @Vininess aw that sucks you didn't get the pc adapter. I got mine with an Xbox so it already had it

**Comment by ajayyy** [comment]
What is kinectPlugin? Did you get the freepie plugin working?

**Comment by BritishLad** [comment]
@ajayyy I got it working with a new installation of freepie, I'll make a video of how I did it cause he didn't explain it very well.

**Comment by BritishLad** [comment]
The math for the hand positions is wrong at the moment thou. I just thought of it

'''
Hydra[0].y = kinectPlugin.handLeftY kinectPlugin.headY ÷ 3 * 2
'''

Takes ur hight ÷ by 3 * 2 to find chest hight. Places hands there

**Comment by Vininess** [comment]
@BritishLad i try to run your code without the Kinect part, just for test, i use the Sixsense_Simple3D to test, i pull the trigger and nothing happens

Because i don't have the kinect cable yet i can't test the complete code

**Comment by BritishLad** [comment]
hmm, i think it might be the buttons code I'm gonna rewrite @ajayyy 's code for my own preference, cause i don't know the layout and then I'm gonna make a map of the buttons and upload it.

**Comment by Vininess** [comment]
Here is the layout of the Vive to Hydra

![razer_hydra_button_mapping](https://user-images.githubusercontent.com/35041490/34775963-32a67198-f5fc-11e7-957c-2e02c3602d9c.png)

I think that is the
B for Trigger
Dpad for TrackPad
A for Press TrackPad
Home For System

And i saw in other codes, minus for recalibrate

**Comment by Vininess** [comment]
@BritishLad and @ajayyy  i didn't get the power supply for the kinect yet, but i have two questions

If i stand my back to the kinect, it can track my hands? assuming my body is between my hand and the sensor?

And  the Kinect one is better than the Kinect 360 for freepie/VR purpose?

**Comment by BritishLad** [comment]
@Vininess you are correct with the sensor and you, as long as it can draw a skeleton around ur head shoulders and it will be able to track ur hands. and the kinect one is better due to its FOV, the one has a higher FOV. it also has better tracking, however kinect 360 does everything u need.

**Comment by Vininess** [comment]
"You are correct with the sensor and you, as long as it can draw a skeleton around ur head shoulders and it will be able to track ur hands"

I see, so the Kinect can presume what my hands should be? no need a second kinect?

"And the kinect one is better due to its FOV, the one has a higher FOV. it also has better tracking, however kinect 360 does everything u need."

I read in Driver4VR forum about the kinect one, and i started thinking: Ok i can do a good job in my room with the FOV of Kinect 360, the play area is about 2 meters per 3 meters, is quiet good i guess

And if Kinect 360 can track the hands and head without lag, it's pointless to get better tracking Kinect one i think, because for our uses makes no diference at all

Is like you try to catch a mouse, you can do with a dynamite, it is powerful but you do the same job using a mouse trap

I thinking right? so the Kinect 360 do the same job in the end?

Thank for the help

**Comment by BritishLad** [comment]
@Vininess yes the kinect 360 does the same as the One. just sees more. and u dont need Driver4VR, Im writing the code for kinect on freepie with wiimote control

**Comment by Vininess** [comment]
@BritishLad I just digging some of his info on Driver4VR foruns, i wait to test and feedback your code, it will be awesome, not just for the price, we will no longer need 3 apps runing to make all work, just your code and the Vridge

I wait and thanks a lot for the answers and support 👍

**Comment by ajayyy** [comment]
Finally got my computer actually working with SteamVR again (updated drivers and windows version and I think SteamVR used to have a bug not supporting my setup well), so I can actually use/test these things. Looking forward to your script :)

**Comment by ajayyy** [comment]
And @Vininess the only issue with kinect is it cannot track your hand if you are covering it, like if you are looking to the side, but that's not too bad.

**Comment by BritishLad** [comment]
hey here's the code made some improvements as such i have yet to add the motion Plus part. @ajayyy you can do it if you like. you guys will need the plugin https://github.com/jakers1403/Kinect-Plugin-Freepie

simple to install just stick the dlls into the plugins folder of freepie and enable them. foreign dlls are not allowed to be ran so enable them in the properties of each dll.

```
def vive_controller1():

	finetuning = 23
	
	hydra[1].x = kinectPlugin.handLeftX*finetuning - kinectPlugin.headX
	hydra[1].y = kinectPlugin.handLeftY*finetuning - kinectPlugin.headY
	hydra[1].z = kinectPlugin.handLeftZ*30 - kinectPlugin.headZ
	
	hydra[1].pitch = -20
	
	Lone = wiimote[1].buttons.button_down(WiimoteButtons.DPadUp) or wiimote[1].buttons.button_down(WiimoteButtons.DPadDown) or wiimote[1].buttons.button_down(WiimoteButtons.DPadLeft) or wiimote[1].buttons.button_down(WiimoteButtons.DPadRight)
	Lfour = wiimote[1].buttons.button_down(WiimoteButtons.Plus)
	Lstart = wiimote[1].buttons.button_down(WiimoteButtons.Home)
	Lbumper = wiimote[1].buttons.button_down(WiimoteButtons.B)
	Ltrigger = wiimote[1].buttons.button_down(WiimoteButtons.A)
	
	hydra[1].one = Lone
	hydra[1].four = Lfour
	hydra[1].start = Lstart
	hydra[1].bumper = Lbumper
	hydra[1].joybutton = Lone
	hydra[1].trigger = Ltrigger
	if wiimote[1].buttons.button_down(WiimoteButtons.DPadRight):
		hydra[1].joyx = 1
	elif wiimote[1].buttons.button_down(WiimoteButtons.DPadLeft):
		hydra[1].joyx = 0
	if wiimote[1].buttons.button_down(WiimoteButtons.DPadUp):
		hydra[1].joyy = 1
	elif wiimote[1].buttons.button_down(WiimoteButtons.DPadDown):
		hydra[1].joyy = 0
	
	hydra[0].x = kinectPlugin.handRightX*finetuning - kinectPlugin.headX
	hydra[0].y = kinectPlugin.handRightY*finetuning - kinectPlugin.headY
	hydra[0].z = kinectPlugin.handRightZ*30 - kinectPlugin.headZ
	
	hydra[0].pitch = -20
		
	Rone = wiimote[0].buttons.button_down(WiimoteButtons.DPadUp) or wiimote[0].buttons.button_down(WiimoteButtons.DPadDown) or wiimote[0].buttons.button_down(WiimoteButtons.DPadLeft) or wiimote[0].buttons.button_down(WiimoteButtons.DPadRight)
	Rfour = wiimote[0].buttons.button_down(WiimoteButtons.Plus)
	Rstart = wiimote[0].buttons.button_down(WiimoteButtons.Home)
	Rbumper = wiimote[0].buttons.button_down(WiimoteButtons.B)
	Rtrigger = wiimote[0].buttons.button_down(WiimoteButtons.A)
	
	hydra[0].one = Rone
	hydra[0].four = Rfour
	hydra[0].start = Rstart
	hydra[0].bumper = Rbumper
	hydra[0].joybutton = Rone
	hydra[0].trigger = Rtrigger
	if wiimote[0].buttons.button_down(WiimoteButtons.DPadRight):
		hydra[0].joyx = 1
	elif wiimote[0].buttons.button_down(WiimoteButtons.DPadLeft):
		hydra[0].joyx = 0
	if wiimote[0].buttons.button_down(WiimoteButtons.DPadUp):
		hydra[0].joyy = 1
	elif wiimote[0].buttons.button_down(WiimoteButtons.DPadDown):
		hydra[0].joyy = 0
	
	joyLi = 0
	joyRi = 1
		
	diagnostics.watch(wiimote[joyLi].buttons.button_down(WiimoteButtons.B))
	diagnostics.watch(wiimote[joyRi].buttons.button_down(WiimoteButtons.B))
	
	diagnostics.watch(wiimote[joyLi].buttons.button_down(WiimoteButtons.A))
	diagnostics.watch(wiimote[joyRi].buttons.button_down(WiimoteButtons.A))
	diagnostics.watch(wiimote[joyRi].buttons.button_down(WiimoteButtons.DPadUp))
	diagnostics.watch(wiimote[joyLi].buttons.button_down(WiimoteButtons.DPadDown))
	
	diagnostics.watch(kinectPlugin.handLeftX)
	diagnostics.watch(kinectPlugin.handLeftY)
	diagnostics.watch(kinectPlugin.handLeftZ)
	
	diagnostics.watch(kinectPlugin.handRightX)
	diagnostics.watch(kinectPlugin.handRightY)
	diagnostics.watch(kinectPlugin.handRightZ)

def hydra_init():
	hydra[0].pitch = 0
	hydra[0].yaw = 0
	hydra[0].roll = 0
	hydra[0].x = 0
	hydra[0].y = 0
	hydra[0].z = 0

	hydra[1].pitch = 0
	hydra[1].yaw = 0
	hydra[1].roll = 0
	hydra[1].x = 0
	hydra[1].y = 0
	hydra[1].z = 0
	
def vive_init():
	global posx
	global posy
	global posz
	posx = 0
	posy = 0
	posz = 0
	global orrpitch
	global orryaw
	global orrroll
	orrpitch = 0
	orryaw = 0
	orrroll = 0
	
global s

if starting:
	vive_init()
	s = 1
	hydra_init()

if s == 1:
	vive_controller1()
	
```

if u guys have motion vr u can add it just rn I dont have mine so i have to wait a bit when i have them i will reupload the code and then everything should work. the finetuning of the program is required. i havent added the math for where your hands need to be so i suggest u guys look into it.

**Comment by ajayyy** [comment]
I don't have the motion plus either. This is really cool I'll test it thanks!

**Comment by Br7an547** [comment]
Hello! Sorry if this completely off-topic, but what exactly does this do and what do I need? I’m very interested in this!

**Comment by BritishLad** [comment]
@Br7an547 its for steam vr. we use freepie a program and 2 wiimotes and a kinect headset as well as phones to make our own vr setup  this is the script we are gunna use for kinect and wiimote compatibility

**Comment by ajayyy** [comment]
@Br7an547 

This repository is code I made a long time ago that is supposed to let you use your hands vr using leap motion for positional tracking and the wiimote for the button presses. (but it is pretty broken)

@BritishLad is working on a more refined script to be able to use the Kinect for positional tracking and the wiimotes for button presses.

We use a program called VRidge to use your phone as a VR headset, and use it in a cardboard style headset, but that only tracks rotation. So for hands and positional tracking you have to do it in other ways.

**Comment by ajayyy** [comment]
Do any of you guys know what YouTube video is linking to my repository, I am very curious about it.

![image](https://user-images.githubusercontent.com/12688112/34911035-6115852e-f88f-11e7-8ee4-f7d3b31d6b83.png)

**Comment by BritishLad** [comment]
@ajayyy ill look into it i think thats how i found it..

**Comment by BritishLad** [comment]
it takes me here https://www.youtube.com/watch?v=VJkkpdSatG4 @ajayyy i feel like he originally used your code as it used to be a link on this video that's how i found it. he recently changed/edited the description.

**Comment by ajayyy** [comment]
Oh, ok. That's pretty cool. Thanks for finding that for me. I left a comment on their linking mine a few days ago thinking it was a different thing haha.

**Comment by ajayyy** [comment]
I can't run 2 Kinect programs at the same time @BritishLad So I can't run open track kinect and your script, however otherwise your script is really cool. Making it track you position would probably be the next step, I'll look into that if you aren't already.

**Comment by BritishLad** [comment]
@ajayyy I haven't found a way to yet transfer the head position into steam vr, if you could find out that would be awesome

**Comment by ajayyy** [comment]
@BritishLad FreePIE supports Freetrack so you can do `freetrack.x` I'm pretty sure. VRidge will send that to SteamVR

**Comment by Vininess** [comment]
Guys, sorry i was out at the weekend

So, i didn't get the kinect cable yet

@BritishLad i test your second code removing the kinect part, for me the trigger button is the A button, instead of normal B button. I can access the sixsens_simple3d but i can't rotate the controller, i have to combine two codes?

@ajayyy the guy of the video is the guy i mentioned that i get the codes to track just the Yaw, Pitch and Roll

@BritishLad have any code to track using ping pong balls to track the head? can the freepie do that? 

@BritishLad and @ajayyy we can use two kinects instead of one to better the track position? or use the kinect to solve the wiiremote drift problem, like use the kinect info to recalibrate the wiiremote?

**Comment by Vininess** [comment]
@Br7an547 we try to figure out how to use the Kinect and wiiremote plus together to play some VR games in steamVR. @BritishLad is writing a code using the FreePie to make with just one click they work together

You nedd just the

Wiiremote with Motion Plus
Kinect for Xbox 360
Smartphone with Vridge installed and a Cardboard

**Comment by ajayyy** [comment]
So, I just checked out Driver4VR, I did not know how powerful it has become, it's a really nice program. Wish it was open source so I could add the wiimote functionality since he said he won't get to it for a few weeks since there are other issues he is fixing now.

**Comment by ajayyy** [comment]
How do you guys teleport in games like the lab, I cannot get it working, is it pressing the grip buttons and pointing it down?

**Comment by Vininess** [comment]
@ajayyy i simple move the left controller point to som place and press the trigger button

**Comment by BritishLad** [comment]
@ajayyy hey mate, just got my motion plus can you help me with the motionplus code as i have tried and Im not getting any output for the pitch and everything.

**Comment by ajayyy** [comment]
https://github.com/ajayyy/Wiimote-FreePIE-VR-Controls/issues/1#issuecomment-330001698

**Comment by BritishLad** [comment]
@ajayyy hey was looking into it and is there a way to use the kinect with freetrack? if so we could use led balls on the wiimotes and track them for better accuracy.

**Comment by ajayyy** [comment]
yea it looks like you can only use the skeleton tracking with freetrack.

**Comment by BritishLad** [comment]
no use led tracking with free track, just use the kinect as the camera

**Comment by ajayyy** [comment]
ooh, I don't think so.

**Comment by TomCallan2** [comment]
@BritishLad or @ajayyy  I know it's a bit of a longshot as it's been a couple of years but any chance either of you still have the kinect plugin? It doesn't seem to be available anywhere.

**Comment by ajayyy** [comment]
> @BritishLad or @ajayyy  I know it's a bit of a longshot as it's been a couple of years but any chance either of you still have the kinect plugin? It doesn't seem to be available anywhere.
> 

Driver4VR? That is another software, not sure if it's still around. Maybe check archive.org? I don't think anyone here ended up making any of their own Kinect software.

I could dig up an old backup if you really need, but it would be an old version

**Comment by TomCallan2** [comment]
@ajayyy You are a genius man, would never have considered archive.org. Chees!

**Comment by ajayyy** [comment]
@TomCallan2 http://www.driver4vr.com/download/ loads for me

**Comment by TomCallan2** [comment]
@ajayyy you might be misunderstanding me, I meant the kinect plugin for freepie, used to be under https://github.com/jakers1403/Kinect-Plugin-Freepie but isn't there any more and it wasn't saved by archive.org

**Comment by LinkON809** [comment]
Hey did anyone get this working?
I've been trying to get my kinect to track a wiimote for vr without paying for driver4vr
and I saw this tread and thought it might be worth it to try it out
